main.go

```diff
@@ -12,23 +12,48 @@ import (
 	"github.com/Yamashou/gqlgenc/generator"
 )
 
-func main() {
-	ctx := context.Background()
-	cfg, err := config.LoadConfigFromDefaultLocations()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%+v", err.Error())
-		os.Exit(2)
-	}
+var generateCmd = &cli.Command{
+	Name:  "generate",
+	Usage: "generate a graphql client based on schema",
+	Flags: []cli.Flag{
+		&cli.StringFlag{Name: "configdir, c", Usage: "the directory with configuration file", Value: "."},
+	},
+	Action: func(ctx *cli.Context) error {
+		configDir := ctx.String("configdir")
+		cfg, err := config.LoadConfigFromDefaultLocations(configDir)
+		if err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "%+v", err.Error())
+			os.Exit(2)
+		}
+
+		clientGen := api.AddPlugin(clientgen.New(cfg.Query, cfg.Client, cfg.Generate))
+		if cfg.Generate != nil {
+			if cfg.Generate.ClientV2 {
+				clientGen = api.AddPlugin(clientgenv2.New(cfg.Query, cfg.Client, cfg.Generate))
+			}
+		}
 
-	clientGen := api.AddPlugin(clientgen.New(cfg.Query, cfg.Client, cfg.Generate))
-	if cfg.Generate != nil {
-		if cfg.Generate.ClientV2 {
-			clientGen = api.AddPlugin(clientgenv2.New(cfg.Query, cfg.Client, cfg.Generate))
+		if err := generator.Generate(ctx.Context, cfg, clientGen); err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "%+v", err.Error())
+			os.Exit(4)
 		}
+		return nil
+	},
+}
+
+func main() {
+
+	app := cli.NewApp()
+	app.Name = "gqlgenc"
+	app.Description = "This is a library for quickly creating strictly typed graphql client in golang"
+	app.Usage = generateCmd.Usage
+	app.DefaultCommand = "generate"
+	app.Commands = []*cli.Command{
+		generateCmd,
 	}
```

Comment:
It seems to me that we should be thinking more about cleaning up the source code here than cleaning up the project directory!

Reply:
can you elaborate please? I don't understand this comment